packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -2163,6 +2165,9 @@ function commitMutationEffects(root: FiberRoot, renderPriorityLevel) {
       if (current !== null) {
         commitDetachRef(current);
       }
+      if (enableScopeAPI && nextEffect.tag === ScopeComponent) {
+        commitAttachRef(nextEffect);
+      }
```

Comment:
Can we separate the feature flag check here?

```suggestion
      if (enableScopeAPI) {
        if (nextEffect.tag === ScopeComponent) {
          commitAttachRef(nextEffect);
        }
      }
```

I find this to be more readable for feature flags.